### DISCUSSION TITLE
Forcing resurveying of a whole area (e.g. after construction)

### DISCUSSION THREAD

**Discussion by peternewman** [streetcomplete/StreetComplete#3651] [Ideas]
> It may be necessary to add minor functionality to trigger resurvey quest older than specific date, rather than relatively to the current date

> I don't see a reason to add a bounding box as long as there is a way (Deeplink?, Setting?) to activate this for all who want to participate. The process would be accompanied by some local communication and probably kick off meeting with introduction in how to do the task.

I've been pondering a more general problem for a while, which I think overlaps with this a lot, hence slightly hijacking this issue.

Consider broadly any type of construction work, they seem to be often knocking down buildings and erecting tower blocks in a lot of areas near me. Or more minor things such as resurfacing roads, or bits in the middle like pedestrianising parts of town (or just giving a bit more space to pedestrians); generally shaking things up significantly over a reasonably large area.

We've currently got a quest where we ask if this construction is finished, but that's basically it.

I think most people would agree that on-the-ground tools like SC (or other mobile apps) are best suited for surveying, otherwise you've essentially got to note down the same info and then process it again when you get home (on iD/JOSM or whatever), which means twice as much work and more risk of error.

Ignoring the practicalities of how it's achieved for the moment, wouldn't it be nice if I could flag up an area within OSM and say they're doing some work on this area until a predicted end date of 1 Jan 2022. I'd then get the construction quest to say have they actually finished. Maybe it overruns a bit, so construction actually finishes a month later (1 Feb 2022); at that point, everything within that area should be resurveyed if it's dated after the 1st Feb.

Because of the above, I'm very behind this idea, but I think it would make sense if it was implemented in such a way it can be used for lots of people, rather than just this specific project; because of the above I think the reality is it's happening all over the place all the time.

Particularly for the more minor pedestrianisation type cases, I don't really see what other option I have but something like this; I don't want to delete all the surface tags, as I'd need to wait until they'd completely finished to ensure I covered them all, but they've already completed some parts, so they could have already been resurveyed.

FWIW, logically to me it would be an area on the OSM map perhaps around the construction tagging; but I realise doing it that way would need a lot wider buy in. As @tordans has sort of alluded to, it would be good if you could do some filtering; e.g. in one case they've only been resurfacing and maybe moving benches etc, but in the towerblock case, whole buildings and paths may be gone and entirely reworked. @tordans request is then just a special case of what you'd do if they'd resurfaced just some roads or similar.

_Originally posted by @peternewman in https://github.com/streetcomplete/StreetComplete/discussions/3644#discussioncomment-1979977_

See also some other related comments in the original discussion too.

**Comment by timothywashere** [comment]
I would like to see an option "This area has changed" that retriggers all quests within a certain distance, say 20m
In my mind this feature is called the holy hand grenade

**Comment by Lee-Carre** [comment]
Interesting. I've had similar thoughts based on similar observations.

For (example) [Moneypenny Lane & Merchants Square](//www.openstreetmap.org/changeset/117269681), since the changes affect the general public (routing, because new road), rather than only folks within the site (a specific building), I gave it more attention & thought.

I leveraged specific tagging (opening_date & start_date), since the estimated completion date was known to inform sensible values for those keys.

I see (at least) 2 parts to this
* identifying *what* needs to be resurveyed
* determining *when* resurveying should be prompted

For *what* to resurvey
* any construction (landuse, building, highway, … I'm sure there are more)
* anything tagged with `fixme(:*)=#SurveyMe` (or similar, including indications of incompleteness which may not be `fixme:*` but `note:*`)
* anything tagged with `*:others=yes` (examples: `payment:*`, `recycling:*`, `fuel:*`)
* anything marked as perma-closed but **not** also `disused:*` (?)
* highways that have had their `smoothness` (or possibly `surface`) changed; for `smoothness` when it's been changed by 2+ steps toward excellent, for `surface` when its been significantly / substantially changed (but, how to define that? Perhaps being changed from any `unpaved` type to any `paved` type?)
* a substantial / large feature has (recently) had its `name` / `brand` (significantly; more than just typo correction) changed (?)

For *when* to prompt
* the existing resurvey intervals for volatile features
* the value of keys like `opening_date` and/or (future) `start_date` since those are (supposed to be) in standardised, machine-readable notation
* if a completion date isn't specified, then a sensible default like ‘every 6 months’

Handling of the case where construction is still on-going is important, I feel, lest the quest become spammy. Maybe that's as simple as updating `check_date` and/or `survey:date` while leaving other tagging as-is.

Also, in my area, there are `landuse=construction` areas which have (verifiably) remained so for *years*. It would be good to have a mechanism to avoid reasking surveyors about something that is unlikely to change in the foreseeable future.

As for related nearby features; could be done by a combination of
* radius from the (newly completed) construction
* the most recent update (based on last-modification date of the element, `check_date`, `survey:date`) being older than when the construction started (which might be tricky to determine, even when querying element history via API) else some typical estimate like ‘6 months ago’
